File: batou-rules/testdata/fixtures/csharp/vulnerable/XssReflected.cs
using System;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace VulnerableApp.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class SearchController : ControllerBase
    {
        // Vulnerable: reflected XSS via Response.WriteAsync
        [HttpGet("search")]
        public async Task SearchAsync(string query)
        {
            var html = "<html><body><h1>Results for: " + query + "</h1></body></html>";
            Response.ContentType = "text/html";
            await Response.WriteAsync(html);
        }

        // Vulnerable: Html.Raw with user input in Razor
        [HttpGet("profile")]
        public IActionResult Profile(string bio)
        {
            ViewBag.UserBio = bio;
            // In the Razor view: @Html.Raw(ViewBag.UserBio)
            return View();
        }

        // Vulnerable: Response.Write with unencoded user input
        [HttpGet("greet")]
        public void Greet(string name)
        {
            Response.Write("<p>Hello, " + name + "!</p>");
        }
    }
}

File: batou-rules/testdata/fixtures/java/vulnerable/XssReflected.java
package com.example.vulnerable;

import java.io.IOException;
import java.io.PrintWriter;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class XssReflected extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String name = request.getParameter("name");
        String query = request.getParameter("q");

        response.setContentType("text/html");
        PrintWriter out = response.getWriter();
        out.println("<html><body>");
        out.println("<h1>Hello, " + name + "</h1>");
        out.println("<p>Search results for: " + query + "</p>");
        out.println("</body></html>");
    }

    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        String feedback = request.getParameter("feedback");
        PrintWriter writer = response.getWriter();
        writer.write("<div class='feedback'>" + feedback + "</div>");
    }
}
